hibernate-core/src/main/java/org/hibernate/loader/Loader.java

```diff
@@ -1666,38 +1663,34 @@ protected void instanceAlreadyLoaded(
 			);
 		}
 
-		if ( persister.getBytecodeEnhancementMetadata().isEnhancedForLazyLoading() && enhancementAsProxyEnabled ) {
-			if ( "merge".equals( session.getLoadQueryInfluencers().getInternalFetchProfile() ) ) {
```

Comment:
This looks a bit puzzling. Do you know why it was limited to "merge" fetch profiles? 

Now you're expanding the set of operations to all, but the test just wants the load. I see we only use this for "merge" and "refresh" BTW... I suppose there's not  something special about "refresh" that we need to take into account during this load opertion?

Reply:
I do not remember the reason why it was limited to the `merge` fetch profile, for the test case the fetch profile value is null but we need to hydrate the entity state and I think anything special is needed for "refresh" if I'm correct it should not reach this point in case of `refresh` for an enhanced proxy (see `DefaultRefreshEventListener#onRefresh` lines 71 to75 )

Reply:
great, thanks